### DISCUSSION TITLE
Packaging into a single .exe file

### DISCUSSION THREAD

**Discussion by JoakimBrandt** [TomSchimansky/CustomTkinter#2836] [Q&A]
Hello,

I am wondering if it is possible to use CustomTkinter for single bundled .exe files? I have tried using the [instructions](https://github.com/TomSchimansky/CustomTkinter/wiki/Packaging) here, but I can't seem to find a way to get my python script into a single .exe file. I want to know if it is an impossible task and I should just use the standard tkinter instead?

**Comment by notluxca** [comment]
I don't know if this is a dumb question of mine, but have u already tried pyinstaller with --onefile command?

**Comment by Akascape** [comment]
@JoakimBrandt 
This may help: https://github.com/TomSchimansky/CustomTkinter/discussions/423#discussioncomment-3549223

**Comment by clxakz** [comment]
> Hello,
> 
> I am wondering if it is possible to use CustomTkinter for single bundled .exe files? I have tried using the [instructions](https://github.com/TomSchimansky/CustomTkinter/wiki/Packaging) here, but I can't seem to find a way to get my python script into a single .exe file. I want to know if it is an impossible task and I should just use the standard tkinter instead?

Use **pyinstaller**

`pyinstaller --noconfirm --onedir --windowed --add-data "C:\Users\name\AppData\Local\Programs\Python\Python311\Lib\site-packages\customtkinter;customtkinter/" yourpythonfile.py`

**Comment by IagoBeuller** [comment]
I was able to do this by updating pyinstaller and also using:
--add-data "G:/Projects/MyApp/venv/Lib/site-packages/customtkinter;customtkinter/"
Updating Pyinstaller: https://www.pythonguis.com/tutorials/packaging-tkinter-applications-windows-pyinstaller/

**Comment by subhradip32** [comment]
Working in an project using customtkinter, I encounter a UI problem in custom tkinter that when I was converted the code to exe using pyinstaller some devices wont able to show few buttons of the main programs do any one know how to solve this issue.
